src/core/WorkflowContext.ts

```diff
@@ -465,7 +465,7 @@ export class WorkFlowContext {
     protected writeSourceMaps(result: any) {
         // Writing sourcemaps
         if (this.sourceMapsProject || this.sourceMapsVendor) {
-            this.output.write(`${this.bundle.name}.js.map`, result.sourceMap);
+            this.output.write(`${this.output.filename}.js.map`, result.sourceMap);
```

Comment:
But when you do bundle splitting because of the `dest` option in the split config, the filename might not be the same as the bundle name.